lib/decorate.js

```diff
@@ -51,10 +51,6 @@ function extendServerError (fn) {
     throw new TypeError('The server error object must be a function')
   }
 
-  if (typeof fn() !== 'object' || fn() === null || Array.isArray(fn())) {
```

Comment:
I didn't understand why you removed this check.
If you pass something that is not an object, strange things will happen.

For example:
```js
const xtend = require('xtend')
xtend({ hello: 'world' }, 'hello') // { '0': 'h', '1': 'e', '2': 'l', '3': 'l', '4': 'o', hello: 'world' }
```

Reply:
But is this a task of the framework? The dev should return an object. When I add this check the function is executed although no error was responded. The dev is forced to make a null check in `extendServerError`

Reply:
Looks like the xtend function is too liberal. We use it only for this case. Why we don't switch to `Object.assign` ?

Reply:
Ok, I've understood your point, we should demand that check to the user, maybe we can be more clear about this in the docs.

Regarding `Object.assign` vs `xtend` we should keep using `xtend`, since it is way faster.